core/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -348,9 +348,10 @@ public SnapshotsInProgress(Entry... entries) {
         return this.entries;
     }
 
-    public Entry snapshot(Snapshot snapshot) {
+    public Entry snapshot(final String repositoryName, final String snapshotName) {
```

Comment:
we can make this `public Entry snapshot(Snapshot snapshot)`
The only place where there is an issue is in the tests. There, we can manually iterate over the entries...